localstack/utils/cloudformation/template_deployer.py

```diff
@@ -41,9 +42,13 @@
 
 ACTION_CREATE = "create"
 ACTION_DELETE = "delete"
-AWS_URL_SUFFIX = "localhost"  # value is "amazonaws.com" in real AWS
+AWS_URL_SUFFIX = "localhost.localstack.cloud"  # value is "amazonaws.com" in real AWS
 IAM_POLICY_VERSION = "2012-10-17"
 
+REGEX_OUTPUT_APIGATEWAY = re.compile(
+    rf"^(https?://.+\.execute-api\.).+-.+-\d\.(amazonaws\.com|{AWS_URL_SUFFIX})/?(.*)$"
```

Comment:
nit: we could replace `.+-` with `[^-]+-`, for slightly stricter matching. (not very critical, though..)

Reply:
changed it to match  2 to 3 segments separated with - followed by a number. should be restricted enough. :sweat_smile: